### ISSUE TITLE
Roslyn custom analyzer unit test problem

### ISSUE THREAD

**Issue by skostic9242** [dotnet/roslyn-analyzers#1239]
Hey,

I wrote custom analyzer which this Initialize method:

```csharp
public override void Initialize(AnalysisContext context)
{
    context.RegisterSyntaxNodeAction(AnalyzeContainer, SyntaxKind.InvocationExpression);
}
```

and analyzer working as expected when i test it manually, but when i try to write unit test for it, i get:
"Assert.IsTrue failed. Mismatch between number of diagnostics returned, expected "1" actual "0"
The problem is on line var diags = compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().Result; 
where diags is empty in this case.
Also, it seems that this is not working when I analyze syntax nodes, when i tried other examples with symbols it worked ok.
I didnt do anything fancy on my own here, just used the template I got with new project creation and replaced it with my code in analyzer and tests. Do I need to do something more in this case or?

Cheers, and tnx in advance for help,
Stefan

**Comment by skostic9242** [comment]
update: Its recognized here in compilationWithAnalyzers.GetAllDiagnosticsAsync().Result 
but GetAnalyzerDiagnosticsAsync().Result is empty...

**Comment by 333fred** [comment]
@skostic9242, rather hard to help you troubleshoot without more context. Any chance you can point us at your code?

**Comment by mavasani** [comment]
@skostic9242 This issue is not related to analyzers in this repo. If you are still having trouble, kindly open a separate issue on https://github.com/dotnet/roslyn/issues.
